### PR TITLE
add coverage tests for probes behavior

### PR DESCRIPTION
**Pull request by matthyx**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Add test coverage on probes before refactoring.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

File: test/e2e_node/container_lifecycle_test.go [hunk 1]
```diff
@@ -746,6 +750,161 @@ var _ = SIGDescribe(framework.WithNodeConformance(), "Containers Lifecycle", fun
...
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Parsing results")
+			podSpec, err = client.Get(context.TODO(), podSpec.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+			results := parseOutput(context.TODO(), f, podSpec)
+
+			ginkgo.By("Analyzing results")
+			// readiness probes are called during pod termination
+			framework.ExpectNoError(results.RunTogether(prefixedName(PreStopPrefix, regular1), prefixedName(ReadinessPrefix, regular1)))
```
Comment (gjkim42) [inline test/e2e_node/container_lifecycle_test.go:837]: @matthyx 
I think we have to give more than 1 second for the preStop hook to deflake [this](https://github.com/kubernetes/kubernetes/issues/124791#issue-2289457502).

(Not sure what's wrong with the OOM case though...)

**Pull request merged**